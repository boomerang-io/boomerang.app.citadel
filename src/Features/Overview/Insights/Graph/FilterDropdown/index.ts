// @ts-expect-error ts-migrate(6142) FIXME: Module './FilterDropdown' was resolved to 'C:/User... Remove this comment to see the full error message
export { default } from "./FilterDropdown"