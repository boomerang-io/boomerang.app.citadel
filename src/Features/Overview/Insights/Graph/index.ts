// @ts-expect-error ts-migrate(6142) FIXME: Module './Graph' was resolved to 'C:/Users/Isabela... Remove this comment to see the full error message
export { default } from "./Graph"