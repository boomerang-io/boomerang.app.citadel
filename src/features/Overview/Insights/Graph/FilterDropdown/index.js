import React, { Component } from "react";
import PropTypes from "prop-types";
import { DropdownV2 } from "carbon-components-react";

export class FilterDropdown extends Component {
  static propTypes = {
    label: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired,
    selectedItem: PropTypes.object
  };

  render() {
    const { selectedItem, onChange, options, label } = this.props;
    return (
      <div style={{ width: "15rem", height: "3rem" }}>
        <DropdownV2 label={label} onChange={onChange} selectedItem={selectedItem} items={options} titleText={label} />
      </div>
    );
  }
}

export default FilterDropdown;
