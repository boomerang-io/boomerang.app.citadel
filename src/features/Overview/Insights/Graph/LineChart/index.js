import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer } from "recharts"; //Removed "Legend"
//import CustomLegend from "./CustomLegend";
import CustomTooltip from "./CustomTooltip";
const colors = ["#047CC0", "#0cd455", "#f39200", "#f0430a", "#8206b3"];

// let lastDots = [];
export default class Example extends PureComponent {
  state = {
    lastDots: [],
    toggledLines: [],
    triggerRerender: []
  };

  static propTypes = {
    chartData: PropTypes.array.isRequired,
    higherValue: PropTypes.number,
    lines: PropTypes.array.isRequired
  };

  componentDidMount() {
    window.addEventListener("resize", this.rerenderLegend);
  }
  componentWillUnmount() {
    window.removeEventListener("resize", this.rerenderLegend);
  }

  rerenderLegend = () => {
    this.setState({ triggerRerender: [] });
  };

  setLastDot = e => {
    const { lastDots } = this.state;
    const hasDot = lastDots.find(dot => dot.key === e.dataKey);
    if (e.index + 1 === this.props.chartData.length && !hasDot) {
      const dot = [].concat(this.state.lastDots, { y: e.cy, key: e.dataKey });
      this.setState({ lastDots: dot });
    }
    if (e.index + 1 === this.props.chartData.length && hasDot && hasDot.y !== e.cy) {
      const newLastDots = lastDots.filter(dot => dot.key !== hasDot.key).concat({ y: e.cy, key: e.dataKey });
      this.setState({ lastDots: newLastDots });
    }
  };

  toggleLine = value => {
    const selectedLine = this.state.toggledLines.find(line => line === value.payload.name);
    if (!selectedLine) {
      this.setState({ toggledLines: this.state.toggledLines.concat(value.dataKey) });
    } else {
      this.setState({ toggledLines: this.state.toggledLines.filter(line => line !== value.payload.name) });
    }
  };

  render() {
    const { chartData, lines } = this.props;
    return (
      <div style={{ display: "flex", alignItems: "center", width: "100%" }}>
        <ResponsiveContainer width="100%" aspect={4 / 0.8} minHeight={100}>
          <LineChart
            width={500}
            height={300}
            data={chartData}
            margin={{
              top: 5,
              right: 90,
              left: 20,
              bottom: 5
            }}
          >
            <defs>
              <filter id="drop-shadow">
                <feGaussianBlur in="SourceAlpha" stdDeviation="3" />
                <feOffset dx="5" dy="5" result="offsetblur" />
                <feFlood flood-color="grey" />
                <feComposite in2="offsetblur" operator="in" />
                <feMerge>
                  <feMergeNode />
                  <feMergeNode in="SourceGraphic" />
                </feMerge>
              </filter>
            </defs>
            <CartesianGrid strokeDasharray="1 0" vertical={false} />
            <XAxis dataKey="date" axisLine={false} tickLine={false} tickMargin={15} />
            {/* <YAxis allowDecimals={false} axisLine={false} tickLine={false} tickMargin={20} domain={[0, higherValue*1.2]} /> */}
            <YAxis allowDecimals={false} axisLine={false} tickLine={false} tickMargin={20} />
            <Tooltip content={<CustomTooltip />} />
            {/* <Legend 
              onClick={this.toggleLine}
              content={
                <CustomLegend
                  toggleItem={this.toggleLine}
                  toggledItems={this.state.toggledLines}
                  lastDots={this.state.lastDots}
                />
              }
              align="right"
              layout="vertical"
              verticalAlign="middle"
              wrapperStyle={{display:"flex", height:"100%", paddingLeft:"2rem"}}
            /> */}
            {lines.map((line, index) => (
              <Line
                dot={this.setLastDot}
                filter="url(#drop-shadow)"
                type="monotone"
                dataKey={!this.state.toggledLines.find(selectedLine => selectedLine === line) ? line : ""}
                name={line}
                key={line}
                stroke={colors[index % colors.length]}
              />
            ))}
          </LineChart>
        </ResponsiveContainer>
      </div>
    );
  }
}
