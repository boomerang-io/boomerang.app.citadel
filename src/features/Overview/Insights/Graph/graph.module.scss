@import "~@boomerang/boomerang-styles/lib/colors";

.container {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  font-weight: 400;
}
.header {
  align-items: center;
  margin-bottom: 2rem;
  display: flex;
  justify-content: space-between;
}
.title {
  margin-left: 1.5rem;
  font-size: 1.25rem;
  font-weight: 500;
}
.filters {
  display: flex;
  justify-items: flex-end;
}
