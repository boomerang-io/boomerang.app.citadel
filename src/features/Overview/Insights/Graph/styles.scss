@import "~@boomerang/boomerang-styles/lib/colors";

.c-graph {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  font-weight: 400;
  .recharts-legend-item {
    margin-bottom: 1rem;
  }
}
.c-graph-header {
  align-items: center;
  margin-bottom: 2rem;
  display: flex;
  justify-content: space-between;
}
.b-graph-header__title {
  margin-left: 1.5rem;
  font-size: 1.25rem;
  font-weight: 500;
}
.c-graph-filters {
  display: flex;
  justify-items: flex-end;
}
