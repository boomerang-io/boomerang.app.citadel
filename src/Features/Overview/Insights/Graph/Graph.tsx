import React, { Component } from "react";
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module '@boo... Remove this comment to see the full error message
import { Tile } from "@boomerang-io/carbon-addons-boomerang-react";
import LineChart from "./LineChart";
// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module './graph.module.scss' or its co... Remove this comment to see the full error message
import styles from "./graph.module.scss";

type Props = {
    formatedData?: any;
};

export class Graph extends Component<Props> {

  state = {
    fromDate: {},
    standard: {},
    toDate: {},
  };

  render() {
    const { formatedData } = this.props;

    return (
      // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
      <Tile
        style={{
          display: "flex",
          flexDirection: "column",
          alignItems: "spaceBetween",
          minHeight: "10rem",
        }}
      >
        {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
        <div className={styles.header}>
          {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
          <h3 className={styles.title}>Violations Trend</h3>
          {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
          <div className={styles.filters}></div>
        </div>
        {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
        <div className={styles.container} data-testid="insights-graph">
          {formatedData.chartData.length > 0 ? (
            // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
            <LineChart
              chartData={formatedData.chartData}
              lines={formatedData.lines}
              higherValue={formatedData.higherValue}
            />
          ) : (
            // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
            <p>No violations data to be found</p>
          )}
        </div>
      </Tile>
    );
  }
}

export default Graph;
