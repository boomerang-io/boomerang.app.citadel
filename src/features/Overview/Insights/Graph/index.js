import React, { Component } from "react";
import PropTypes from "prop-types";
import { Tile } from "carbon-components-react";
import LineChart from "./LineChart";
import styles from "./graph.module.scss";

export class Graph extends Component {
  static propTypes = {
    formatedData: PropTypes.object
  };

  state = {
    fromDate: {},
    standard: {},
    toDate: {}
  };

  render() {
    const { formatedData } = this.props;

    return (
      <Tile
        style={{
          width: "100%",
          display: "flex",
          flexDirection: "column",
          alignItems: "spaceBetween",
          minHeight: "10rem"
        }}
      >
        <div className={styles.header}>
          <h3 className={styles.title}>Violations Trend</h3>
          <div className={styles.filters}></div>
        </div>
        <div className={styles.container} data-testid="insights-graph">
          {formatedData.chartData.length > 0 ? (
            <LineChart
              chartData={formatedData.chartData}
              lines={formatedData.lines}
              higherValue={formatedData.higherValue}
            />
          ) : (
            <p>No violations data to be found</p>
          )}
        </div>
      </Tile>
    );
  }
}

export default Graph;
