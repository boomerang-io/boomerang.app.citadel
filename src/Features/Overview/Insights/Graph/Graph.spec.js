import React from "react";
import Graph from "./index";

jest.mock("./LineChart", () => "LineChart");

const formatedData = {
  chartData:[{date:1558457676, test1:1, test2: 2}],
  lines:["test1", "test2"],
  higherValue: 2
};

describe("Graph --- Snapshot", () => {
  it("Capturing Snapshot of Graph", async () => {
    const { baseElement } = global.rtlRender(<Graph formatedData={formatedData} />);
    expect(baseElement).toMatchSnapshot();
  });
});
