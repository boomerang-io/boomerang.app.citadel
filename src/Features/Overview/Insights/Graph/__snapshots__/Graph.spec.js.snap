// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph --- Snapshot Capturing Snapshot of Graph 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="bx--tile"
      style="width: 100%; display: flex; flex-direction: column; align-items: spaceBetween; min-height: 10rem;"
    >
      <div
        class="header"
      >
        <h3
          class="title"
        >
          Violations Trend
        </h3>
        <div
          class="filters"
        />
      </div>
      <div
        class="container"
        data-testid="insights-graph"
      >
        <linechart
          chartdata="[object Object]"
          highervalue="2"
          lines="test1,test2"
        />
      </div>
    </div>
  </div>
</body>
`;
